src/fde.h

```diff
@@ -84,8 +84,25 @@ class fde
     /// use I/O methods that maintain an internal-to-them buffer
     void useBufferedIo(READ_HANDLER *, WRITE_HANDLER *);
 
-    int read(int fd, char *buf, int len) { return readMethod_(fd, buf, len); }
-    int write(int fd, const char *buf, int len) { return writeMethod_(fd, buf, len); }
+    /// call the read method set by fde::setIo() requesting len more bytes.
+    /// Performs I/O counting for all bytes received.
+    /// \returns the number of bytes actually received
```

Comment:
This function does not always return "the number of bytes actually received"

```suggestion
    /// \returns the read method result
```